### PR TITLE
port exercise bowling from JS track - fixes #332

### LINKED ISSUE #332: Port bowling exercise from JavaScript track
As part of the job to make the track compatible with *Nextercism*, we identified there are missing exercises in this track that are present in the [JavaScript](https://github.com/exercism/javascript) track. One of them is [bowling](https://github.com/exercism/javascript/tree/master/exercises/bowling).

The task would be to add the `bowling` exercise to this track, copying files from the JavaScript track and adapting the code to the ECMAScript 6 standard.

Basic steps would include:

1. Copy all files from `<javascript repo>/exercises/bowling` (https://github.com/exercism/javascript) to `<ecmascript repo>/exercises/bowling` (https://github.com/exercism/ecmascript)
2. Copy the main `package.json` file (it's in the root folder of the `ecmascript` repo https://github.com/exercism/ecmascript/blob/master/package.json) to the `<ecmascript repo>/exercises/bowling` folder
3. Adapt the code in `bowling.spec.js` file to use ECMAScript 6 features (`import` modules, arrow functions, `const` and/or `let`, any other feature you think will make the code better). Additionally, the JavaScript track uses Jasmine where the ECMAScript track uses Jest.  So, `it` and `xit` must be *translated* to `test` and `xtest`. Feel free to adapt `example.js` as well, just for fun (as long as tests pass)
4. In the `README.md` of the exercise, replace sections from *Setup* and below (Setup, Making the test suite pass, Source and Submitting incomplete solutions) with similar sections from any ECMAScript exercise, for example [`accumulate` README](https://github.com/exercism/ecmascript/blob/master/exercises/accumulate/README.md). They'll include Setup, Requirements, Making the test suite pass, Source and Submitting incomplete solutions.
5. Copy some exercise information from [JavaScript `config.json` file](https://github.com/exercism/javascript/blob/master/config.json) to the [ECMAScript `config.json` file](https://github.com/exercism/ecmascript/blob/master/config.json). This information is an object in the `exercises` array property with a `slug` property equals to `bowling`. Copy that object and place it in the same (or most similar) position in the ECMAScript `config.json` file. Both `config.json` files are in the root folder of the repo.

@exercism/ecmascripts - Do you miss something here?

### PR DESCRIPTION
**Pull request by trvrfrd**
I noticed there was a stale PR for porting this exercise and decided to give it a shot.

I think I covered everything mentioned in #332, but let me know if I missed anything or you have any suggestions for improvement! I tried to keep everything as close to the original JavaScript exercise as possible, just updating things to ES6 syntax where it seemed appropriate.

I also noticed that the description of the code in the Requirements section doesn't exactly match the interface expected by the specs, but I decided to leave it as-is for now.

### CODE REVIEW

**Comment by matthewmorgan** [comment]
@trvrfrd the CI build failed, because the UUID is not unique- it looks like you re-used the JavaScript one?  You'll want to generate a [new one](https://gist.github.com/jed/982883):

```
(function b(a){return a?(a^Math.random()*16>>a/4).toString(16):([1e7]+-1e3+-4e3+-8e3+-1e11).replace(/[018]/g,b)})()
```

I'll look over the PR and give you feedback.  Thanks for doing this!

File: config.json [hunk 1]
```diff
@@ -950,6 +950,22 @@
       ]
     },
     {
+      "core": false,
+      "difficulty": 8,
+      "slug": "bowling",
+      "topics": [
+        "arrays",
+        "control-flow-(conditionals)",
```
Comment (matthewmorgan) [inline config.json]: The topic names should all be capitalized.

File: exercises/bowling/README.md [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+
+Score a bowling game.
+
+Bowling is game where players roll a heavy ball to knock down pins
+arranged in a triangle. Write code to keep track of the score
+of a game of bowling.
+
+## Scoring Bowling
+
+The game consists of 10 frames. A frame is composed of one or two ball throws with 10 pins standing at frame initialization. There are three cases for the tabulation of a frame.
```
Comment (matthewmorgan) [inline exercises/bowling/README.md]: "throws" should be "thrown."

File: exercises/bowling/README.md [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+* An open frame is where a score of less than 10 is recorded for the frame. In this case the score for the frame is the number of pins knocked down.
+
+* A spare is where all ten pins are knocked down after the second throw. The total value of a spare is 10 plus the number of pins knocked down in their next throw.
```
Comment (matthewmorgan) [inline exercises/bowling/README.md]: Perhaps this should finish "number of pins knocked down in the _second_ throw."  At the least "their" should be replaced with "the."

File: exercises/bowling/README.md [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+
+* A strike is where all ten pins are knocked down after the first throw. The total value of a strike is 10 plus the number of pins knocked down in their next two throws. If a strike is immediately followed by a second strike, then we can not total the value of first strike until they throw the ball one more time.
```
Comment (matthewmorgan) [inline exercises/bowling/README.md]: Again, not "_their+ next two throws"-- "_the_ next two throws."  Also, the use of "they throw" in the last sentence should probably be replaced with "the player throws."

**Comment by trvrfrd** [comment]
I didn't even think about the UUID, and I definitely just pasted the README
over without giving it a close read. I'm at work right now but I'll make
the fixes when I get home this evening. Thanks for the feedback!

On Oct 12, 2017 7:09 AM, "Matthew Morgan" <notifications@github.com> wrote:

*@matthewmorgan* requested changes on this pull request.

@trvrfrd <https://github.com/trvrfrd> there are a few things in the README
that should be corrected, mostly grammar issues. I imagine these are
original from the JS version, so I suppose I should have a closer look at
that one!

Please let me know if you have any questions. Thanks!
------------------------------

In config.json
<https://github.com/exercism/ecmascript/pull/355#discussion_r144256249>:

> @@ -950,6 +950,22 @@
       ]
     },
     {
+      "core": false,
+      "difficulty": 8,
+      "slug": "bowling",
+      "topics": [
+        "arrays",
+        "control-flow-(conditionals)",

The topic names should all be capitalized.
------------------------------

In exercises/bowling/README.md
<https://github.com/exercism/ecmascript/pull/355#discussion_r144256368>:

> @@ -0,0 +1,80 @@
+# Bowling
+
+Score a bowling game.
+
+Bowling is game where players roll a heavy ball to knock down pins
+arranged in a triangle. Write code to keep track of the score
+of a game of bowling.
+
+## Scoring Bowling
+
+The game consists of 10 frames. A frame is composed of one or two
ball throws with 10 pins standing at frame initialization. There are
three cases for the tabulation of a frame.

"throws" should be "thrown."
------------------------------

In exercises/bowling/README.md
<https://github.com/exercism/ecmascript/pull/355#discussion_r144256548>:

> @@ -0,0 +1,80 @@
+# Bowling
+
+Score a bowling game.
+
+Bowling is game where players roll a heavy ball to knock down pins
+arranged in a triangle. Write code to keep track of the score
+of a game of bowling.
+
+## Scoring Bowling
+
+The game consists of 10 frames. A frame is composed of one or two
ball throws with 10 pins standing at frame initialization. There are
three cases for the tabulation of a frame.
+
+* An open frame is where a score of less than 10 is recorded for the
frame. In this case the score for the frame is the number of pins
knocked down.
+
+* A spare is where all ten pins are knocked down after the second
throw. The total value of a spare is 10 plus the number of pins
knocked down in their next throw.

Perhaps this should finish "number of pins knocked down in the *second*
throw." At the least "their" should be replaced with "the."
------------------------------

In exercises/bowling/README.md
<https://github.com/exercism/ecmascript/pull/355#discussion_r144256905>:

> +
+Score a bowling game.
+
+Bowling is game where players roll a heavy ball to knock down pins
+arranged in a triangle. Write code to keep track of the score
+of a game of bowling.
+
+## Scoring Bowling
+
+The game consists of 10 frames. A frame is composed of one or two
ball throws with 10 pins standing at frame initialization. There are
three cases for the tabulation of a frame.
+
+* An open frame is where a score of less than 10 is recorded for the
frame. In this case the score for the frame is the number of pins
knocked down.
+
+* A spare is where all ten pins are knocked down after the second
throw. The total value of a spare is 10 plus the number of pins
knocked down in their next throw.
+
+* A strike is where all ten pins are knocked down after the first
throw. The total value of a strike is 10 plus the number of pins
knocked down in their next two throws. If a strike is immediately
followed by a second strike, then we can not total the value of first
strike until they throw the ball one more time.

Again, not "_their+ next two throws"-- "*the* next two throws." Also, the
use of "they throw" in the last sentence should probably be replaced with
"the player throws."

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/exercism/ecmascript/pull/355#pullrequestreview-68887207>,
or mute the thread

**Comment by matthewmorgan** [comment]
@trvrfrd somehow I missed the notification that you had done this work.  Nice job!

Can you squash your commits to a single commit?  If you need help with this, lmk.

Otherwise this is ready to merge!

**Comment by trvrfrd** [comment]
@matthewmorgan No worries! I kind of forgot about it myself :eyes: Just pushed up the squashed version, thanks again for your help!

**Pull request merged**